lib/omniauth/strategies/openid_connect.rb

```diff
@@ -203,7 +203,9 @@ def client_options
       end
 
       def new_state
-        state = options.state.call if options.state.respond_to? :call
+        if options.state.respond_to? :call
+          state = options.state.arity == 1 ? options.state.call(env) : options.state.call
```

Comment:
1. shouldn't you check arity of `call` with `options.state.method(:call).arity == 1` ?
2. please consider change `state` variable definition to something like this:

```ruby
state = if options.state.respond_to?(:call)
          if options.state.method(:call).arity == 1
            options.state.call(env)
          else
            options.state.call
          end
        end
```

It's more verbose but way more readable.

Reply:
It won't work quite like that, but we could eliminate the ternary operator.

options.state is a method, so it does not exactly have methods.
You can see the arity usage examples at https://ruby-doc.org/core-2.6.1/Method.html#method-i-arity

The ternary operator is suggested by rubocop (search for ternary on https://github.com/rubocop-hq/ruby-style-guide ) but does not make much difference here and your suggestions is clean.

The precise formatting is your call, as it is your project. Uniformity and maintainability are significant concerns for a project.